servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -2356,7 +2356,7 @@ public final Completable flatMapCompletableDelayError(Function<? super T, ? exte
      */
     @Deprecated
     public final Publisher<T> multicastToExactly(int expectedSubscribers, int queueLimit) {
-        return new MulticastPublisher<>(this, expectedSubscribers, true, queueLimit, t -> completed());
+        return new MulticastPublisher<>(this, expectedSubscribers, true, true, queueLimit, t -> completed());
```

Comment:
I know this would be a backward incompatible change but using `false` as default would be better. Typically, multicast is used for sources that do not otherwise support multiple subscribes (as our processors do), so inadvertently subscribing multiple times (due to the lifetime of downstream subscribers) to upstream produces hard to debug issues.

Reply:
I would prefer to keep the default the same in this PR. It isn't clear which default behavior is better... I can imagine not propagating cancellation resulting in a hang which would also be hard to debug. At least propagating the cancel will ensure definitive termination.

Reply:
That is fine; IMHO preserving lifecycle of upstream is an appropriate default behavior. Multicast is providing just a way to support multiple subscribers. If upstream is an infinite stream, it will stay infinite with multicast else it will terminate with upstream.